### PR TITLE
RSDK-2488-add-final-slam-api-methods

### PR DESCRIPTION
**Pull request by nicksanford**
[Ticket](https://viam.atlassian.net/browse/RSDK-2488)

Add:
1. GetPosition
2. GetPointCloudMap
3. GetInternalState
& associated message types

Deprecate:
1. GetPositionNew
2. GetPointCloudMapStream
3. GetInternalStateStream
& associated message types

### CODE REVIEW

**Review by tessavitabile — APPROVED** [review]

File: component/audioinput/v1/audioinput.pb.gw.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 
 	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
 	"github.com/grpc-ecosystem/grpc-gateway/v2/utilities"
-	"go.viam.com/api/common/v1"
+	v1_0 "go.viam.com/api/common/v1"
```
Comment (EshaMaharishi) [inline component/audioinput/v1/audioinput.pb.gw.go:18]: Should these changes be showing up in this PR?

**Review by EshaMaharishi — COMMENTED** [review]
Quick question before reviewing slam.proto

**Review by EshaMaharishi — APPROVED** [review]
Changes to slam.proto lgtm

File: component/audioinput/v1/audioinput.pb.gw.go (hunk 1 above)
Comment (nicksanford) [reply component/audioinput/v1/audioinput.pb.gw.go:18]: The answer appears to be yes as that was done by the CI system: https://github.com/viamrobotics/api/pull/226/commits

I also confirmed with james here: https://viaminc.slack.com/archives/C01Q5S8V6TB/p1680025534478849?thread_ts=1680018444.526449&cid=C01Q5S8V6TB

Comment (EshaMaharishi) [reply component/audioinput/v1/audioinput.pb.gw.go:18]: Ah I see, it gets added by CI to the PR itself after (one?) approval. Thank you!

**Review by JohnN193 — APPROVED** [review]

**Pull request merged**